drivers/timer/stm32_radio_timer.c

```diff
@@ -0,0 +1,203 @@
+/* stm32_radio_timer.c - STM32WB0x Radio Timer */
+
+/*
+ * Copyright (c) 2025 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/timer/system_timer.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/spinlock.h>
+#include <cmsis_core.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/drivers/counter.h>
+#include "stm32wb0x_hal_radio_timer.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(radio_timer_driver);
+
+/* Max HS startup time expressed in system time (1953 us / 2.4414 us) */
+#define MAX_HS_STARTUP_TIME	320
+#define BLE_WKUP_PRIO		0
+#define BLE_WKUP_FLAGS		0
+#define CPU_WKUP_PRIO		1
+#define CPU_WKUP_FLAGS		0
+#define RADIO_TIMER_ERROR_PRIO	3
+#define RADIO_TIMER_ERROR_FLAGS	0
+
+#define MULT64_THR_FREQ		(806)
+#define TIMER_MAX_VALUE		(0xFFFFFFFFU)
+#define TIMER_WRAPPING_MARGIN	(4096)
+#define MAX_ALLOWED_DELAY	(TIMER_MAX_VALUE - TIMER_WRAPPING_MARGIN)
+
+
+BUILD_ASSERT(DT_NODE_HAS_STATUS(DT_NODELABEL(clk_lsi), disabled),
+	     "At the moment, LSI is not supported");
```

Comment:
Nit:
```suggestion
	     "LSI not supported yet");
```
\+ we should check if the `slow-clock` property on `rcc` is `clk_lsi` instead... but that's fine.